examples/private_information_retrieval/module/rust/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Private Information Retrieval example for Project Oak.
+//!
+//! This shows how an Oak Node can host an in-memory database and process private queries from
+//! clients.
+//!
+//! Current example implementation uses a list of Santander Cycles in London as test database:
+//! https://tfl.gov.uk/tfl/syndication/feeds/cycle-hire/livecyclehireupdates.xml
+//!
+//! Clients send their location coordinates (latitude and longitude) and the Oak Node returns the
+//! location of the closest Point Of Interest (in our case a bike station).
+
+mod proto {
+    include!(concat!(
+        env!("OUT_DIR"),
+        "/oak.examples.private_information_retrieval.rs"
+    ));
+}
+
+mod database;
+#[cfg(test)]
+mod tests;
+
+use database::load_database;
+use log::{debug, error};
+use oak::grpc;
+use proto::{
+    Location, PointOfInterest, PrivateInformationRetrieval, PrivateInformationRetrievalDispatcher,
+};
+
+/// Oak Node that contains an in-memory database of Points Of Interest.
+pub struct PrivateInformationRetrievalNode {
+    points_of_interest: Vec<PointOfInterest>,
+}
+
+impl PrivateInformationRetrievalNode {
+    /// Returns a Euclidean distance between two locations.
```

Comment:
What's the unit of this distance? I guess degrees?

Reply:
Personally I think converting to km is complete overkill, especially since AFAICT you are not actually using the absolute value for anything, you just want to find the minimum one. Or am I missing something?